hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSecondaryIndexPruning.scala

```diff
@@ -62,6 +62,12 @@ import scala.concurrent.{Await, ExecutionContext, Future}
 @Tag("functional")
 class TestSecondaryIndexPruning extends SparkClientFunctionalTestHarness {
 
+  def repeatTest(times: Int)(test: => Unit): Unit = {
+    for (_ <- 1 to times) {
+      test
+    }
+  }
```

Comment:
I guess this was for local testing? If so, please remove it from this PR. I think it would be better if we add retry api in some super class so that devs can easily run scala test N times. But, that can be in a separate PR.

Reply:
I had two separate PRs mixed here. Reverted the other one.
Will create a separate PR for the retry mechanism.

Reply:
Created https://github.com/apache/hudi/pull/12691/files, PTAL

Reply:
Cool. Left a comment on #12691